### PR TITLE
COP:11940/12292/11996: RCCU/non-RCCU airpax target view

### PR DESCRIPTION
**Pull request by adam-thorpe**
Features:
- Add assign button on the receipted tab
- Redirects user to Assign Page
- Form with autocomplete box that will grab list of emails from identity service
- Submitting this form will redirect the user back to the airpax target page and move target to assigned tab

To Do:
- Add tests

### CODE REVIEW

File: client/src/pages/targets/utils/CommonUtil.js [hunk 1]
```diff
@@ -9,6 +9,23 @@ const getView = () => {
   return localStorage.getItem(LS_KEYS.VIEW);
 };
 
+const getPathByView = (view) => {
```
Comment (charles474) [inline client/src/pages/targets/utils/CommonUtil.js:12]: Test missing for this function.

File: client/src/pages/targets/v2/Assign.jsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
+import axios from 'axios';
```
Comment (charles474) [inline client/src/pages/targets/v2/Assign.jsx:1]: Test(s) missing for this Component.

File: client/src/pages/targets/components/TargetListItemOptions.jsx [hunk 3]
```diff
@@ -9,11 +9,34 @@ import { TARGET_STATUS, TASK_STATUS } from '../../../utils/constants';
 
 import { getView, isViewAirpax } from '../utils/CommonUtil';
 
+export const RCCUReceiptButton = ({ selectedTab, targetStatus, onClick, permission }) => {
```
Comment (charles474) [inline client/src/pages/targets/components/TargetListItemOptions.jsx:12]: Test(s) missing for this button.

File: client/src/pages/targets/components/TargetListItemOptions.jsx [hunk 4]
```diff
@@ -55,6 +78,29 @@ export const RoRoReceiptButton = ({ selectedTab, targetStatus, onClick, permissi
   return null;
 };
 
+export const AssignButton = ({ selectedTab, targetStatus, onClick, permission }) => {
```
Comment (charles474) [inline client/src/pages/targets/components/TargetListItemOptions.jsx:81]: Test(s) missing for this button.

File: client/src/pages/targets/components/TargetListCard.jsx [hunk 5]
```diff
@@ -26,11 +26,13 @@ const TargetListCard = ({
   menuId,
   tableHeaders,
   groupedTargets,
+  handleOnRCCUReceipt,
```
Comment (charles474) [inline client/src/pages/targets/components/TargetListCard.jsx:29]: No test that simulates the actioning of this function.

File: client/src/pages/targets/components/TargetListCard.jsx [hunk 6]
```diff
@@ -26,11 +26,13 @@ const TargetListCard = ({
...
   handleOnReceipt,
   handleOnAccept,
   handleRadioButton,
   handleViewDetails,
   handleRecordOutcome,
+  handleAssign,
```
Comment (charles474) [inline client/src/pages/targets/components/TargetListCard.jsx:35]: Same as above.

File: client/src/pages/targets/forms/cop-assign.json [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+            "add": "Add another officer"
+          },
+          "item": [
+            {
+              "id": "email",
+              "fieldId": "email",
+              "label": "Enter name or email address, of officer",
+              "type": "autocomplete",
+              "required": true,
+              "clearable": true,
```
[the remark this answers is not in the document]
Comment (adam-thorpe) [reply client/src/pages/targets/forms/cop-assign.json:29]: I'm confused, what's the reason for this? Are we not using the json to define the form anymore, but instead use the components directly?

[the remark this answers is not in the document]
Comment (charles474) [reply client/src/pages/targets/forms/cop-assign.json:29]: We are still use the JSON but the component which that particular component is referring to is not one supported by default by the form renderer so we have to provide it in that hook.

File: client/src/pages/targets/v2/Assign.jsx [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+  const [canAssign, setCanAssign] = useState({
+    allowed: false,
+    reason: `${t('error.page-not-found.title')}`,
+  });
+
+  const onGetComponent = (component, wrap) => {
+    const attrs = Utils.Component.clean(component, [
+      'clearable',
+      'fieldId',
+      'dynamicOptions',
```
Comment (charles474) [inline client/src/pages/targets/v2/Assign.jsx:40]: We can remove this `dynamicOptions` value. This was only used on the targeting UI to trigger setting up filter counts.

**Review by charles474 — APPROVED** [review]
LGTM

**Pull request merged**